app/src/sync/process-initialization.ts

```diff
@@ -69,6 +59,35 @@ export async function update_directory() {
   ensureActiveProjects();
 }
 
+/**
+ * Create and store a listings object for this conductor instance
+ *
+ * @param url - URL of the Conductor instance
+ * @returns A Listings object
+ */
+async function generate_listing(url: string) {
+  const url_object = new URL(url);
+  const listing: ListingsObject = {
+    _id: url_object.host,
+    conductor_url: url,
+    name: url_object.host, // default name from hostname
+    description: '',
+  };
+
+  await fetch(url + '/api/info')
+    .then(response => response.json())
+    .then(data => {
+      listing.name = data.name;
+      listing.description = data.description;
+    })
+    .catch(() => {
+      listing.description = 'No Description';
+    });
+
+  directory_db.local.put(listing);
```

Comment:
What happens if this function hangs or errors? noting that we've lost the benefits of async in this code.